packages/components/src/internal/query/filter.ts

```diff
@@ -47,3 +50,96 @@ export function isEqual(first: List<Filter.IFilter>, second: List<Filter.IFilter
 
     return isEqual;
 }
+
+function getColumnSelect(columnName: string): string {
+    const columnNameParts = columnName.split('/');
+    const formattedParts = [];
+    columnNameParts.forEach(part => {
+        if (part) {
+            formattedParts.push('"' + part.replace('"', '""') + '"');
+        }
+    });
+
+    return formattedParts.join('.');
+}
+
+function getLabKeySqlValue(value: any, jsonType: JsonType): any {
+    if (jsonType === 'string' || jsonType === 'date') {
+        return "'" + value.toString().replace("'", "''") + "'";
+    }
+
+    if (jsonType === 'boolean')
+        return value?.toLowerCase() === 'true' || value?.toLowerCase() === 'yes' || value?.toLowerCase() === 'on'
+            ? 'TRUE'
+            : 'FALSE';
+
+    return value;
+}
+
+/**
+ * Note: this is an experimental API that may change unexpectedly in future releases.
+ * From a filter and its column jsonType, return the LabKey sql operator clause
+ * @param filter The Filter
+ * @param jsonType The json type ("string", "int", "float", "date", or "boolean") of the field
+ * @return labkey sql fragment
+ */
+export function getLabKeySql(filter: Filter.IFilter, jsonType: JsonType): string {
+    const filterType = filter.getFilterType();
+
+    const columnNameSelect = getColumnSelect(filter.getColumnName());
+
+    let operatorSql = null;
+
+    if (filterType.getURLSuffix() === Filter.Types.HAS_ANY_VALUE.getURLSuffix()) return null;
+
+    if (filterType.getLabKeySqlOperator()) {
+        if (!filterType.isDataValueRequired()) operatorSql = filterType.getLabKeySqlOperator();
+        else operatorSql = filterType.getLabKeySqlOperator() + ' ' + getLabKeySqlValue(filter.getValue(), jsonType);
+    } else if (filterType.isMultiValued()) {
+        const values = filterType.parseValue(filter.getValue());
+
+        if (
+            filterType.getURLSuffix() === Filter.Types.IN.getURLSuffix() ||
+            filterType.getURLSuffix() === Filter.Types.NOT_IN.getURLSuffix()
```

Comment:
Do we have to compare the URLSuffixes here? Can we not compare the types? (Or is this just a nod toward efficiency?)

Reply:
Multiple filter types can be the same filter but are just different instances, for example, In and EQUALS_ONE_OF